packages/core/src/diff/field.ts

```diff
@@ -68,6 +68,7 @@ export function changesInField(
     },
   });
 
+  // TODO: Heshan -> Introduce a mutual option to this function to detect changes in directive usage
```

Comment:
I've added this TODO comment to everyplace I could find that we need to call `directiveUsageChanged` function defined in `directive-usage.ts` file. 

A good example would be what's done in `packages/core/src/diff/enum.ts` already.